pkg/agent/run.go

```diff
@@ -131,6 +143,7 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 	systemd.SdNotify(true, "READY=1\n")
 
 	<-ctx.Done()
+	wg.Wait()
```

Comment:
Imagine something goes wrong in netpol and we get `ctx.Done()`. Could it be that flannel still runs and never executes `wg.Done()` and thus we stay stuck here?

Reply:
Yes, that might be the case. I'm wondering if we should pass the `stopChan` to all goroutines, including the one with flannel and close that channel when after `<-ctx.Done` and then `wg.Wait()`. WDYT?

That would mean killing the whole agent when one component dies, which seems like was never done, but I think that's a good way of handling failures.

Or if we don't want to end k3s execution because of one component - we would need to come up with some recovery logic.

Reply:
The problem I see is that the code of flannel with backend vxlan is not executing any `<-ctx.Done()`, so even if ctx finishes, I don't think the goroutine has a way to "exit" properly

Reply:
That doesn't sound good. I also see that in case of flannel failure we just log it without any other reaction. 

https://github.com/k3s-io/k3s/blob/5acd0b9008c93e9ce64e7b7beeabda9248408374/pkg/agent/flannel/setup.go#L100-L105

Should we somehow change it?

Reply:
This isn't a log-and-ignore; `Fatalf()` will cause the k3s process to exit.

Reply:
So you would be fine with the current code, but without `wg.Wait()`?

Reply:
@brandond ping, please let me know what you think ^^